### PR TITLE
Note and Modal component

### PR DESCRIPTION
**Pull request by Muzammil-khan710**
In this PR, I have added modal and note components
Note Context
Private and Auth routes
Adding a card with API call to the backend

### CODE REVIEW

**Review by Shubhamkr2610 — APPROVED** [review]
Code looks good to me . Easy to read and understand . You can go for merge

File: src/components/Note/note.css [hunk 1]
```diff
@@ -15,6 +15,7 @@
     border-radius: 8px;    
     padding: 1rem;
     background-color: #f7f7f7;
```
Comment (zoyatanweer) [inline src/components/Note/note.css]: use var for color

File: src/components/Sidebar/sidebar.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.side-container {
+    display: flex;
+    flex-direction: column;
+    gap: 3rem;
+    padding: 3rem;
+    height: 100vh;
+    width: 3rem;
+    border-right: 2px solid green;
```
Comment (zoyatanweer) [inline src/components/Sidebar/sidebar.css]: use var for color

**Review by zoyatanweer — APPROVED** [review]
- code looks clean and good.
- great UI
- good to merge

**Pull request merged**